File: solutions/r/raindrops/1/raindrops.R
raindrops <- function(number) {
  drops <- list('3' = "Pling", '5' = "Plang", '7' = "Plong")
  result <- ""

  for (key in names(drops)) {
    if (number %% as.integer(key) == 0) {
      result <- paste0(result, drops[[key]])
    }
  }

  if (nchar(result) == 0) {
    return(as.character(number))
  } else {
    return(result)
  }
}

File: solutions/javascript/raindrops/4/raindrops.js
//
// This is only a SKELETON file for the 'Raindrops' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const DROPS = {
  3: 'Pling',
  5: 'Plang',
  7: 'Plong'
}
export const convert = num => {
  return Object.keys(DROPS)
          .reduce((acc, curr) => {
            return num % curr === 0 ? acc + DROPS[curr] : acc
          }, '') || `${num}`
}